packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/BarChartViewer/BarChartViewerContainer.spec.js

```diff
@@ -143,60 +133,55 @@ describe('Component > BarChartViewerContainer', function () {
         .reply(200, nextSubjectJSON)
     })
 
-    afterEach(function () {
-      cdmSpy.resetHistory()
-      cduSpy.resetHistory()
-    })
-
     after(function () {
-      cdmSpy.restore()
-      cduSpy.restore()
       nock.cleanAll()
       nockScope.persist(false)
     })
 
     it('should set the component state with the json data', function (done) {
-      wrapper = shallow(
-        <BarChartViewerContainer
-          subject={subject}
-        />
-      )
-
-      expect(wrapper.state().JSONdata).to.be.empty()
-      cdmSpy.returnValues[0].then(() => {
-        expect(wrapper.state().JSONdata).to.deep.equal(subjectJSON)
-      }).then(done, done)
-    })
-
-    it('should call the onReady prop', function (done) {
-      const onReadySpy = sinon.spy()
-      wrapper = shallow(
+      const onReady = sinon.stub().callsFake(() => {
+        wrapper.update()
+        const bcv = wrapper.find(BarChartViewer)
+        const { data, chartOptions } = subjectJSON
+        const { margin, xAxisLabel, yAxisLabel } = chartOptions
+      
+        expect(bcv.props()).to.deep.equal({ data, margin, xAxisLabel, yAxisLabel })
+        done()
+      })
+      const onError = sinon.stub().callsFake(() => {
+        expect.fail('should not call onError')
+        done()
+      })
+      wrapper = mount(
         <BarChartViewerContainer
-          onReady={onReadySpy}
+          onError={onError}
+          onReady={onReady}
           subject={subject}
         />
       )
-
-      cdmSpy.returnValues[0].then(() => {
-        expect(onReadySpy).to.have.been.calledOnceWith({ target: wrapper.instance().viewer.current })
-      }).then(done, done)
     })
 
     it('should update component state when there is a new valid subject', function (done) {
```

Comment:
This test isn't directly testing component state any more. It's testing the data props that are passed down to the subject viewer from the container.